### PR TITLE
[FEATURE] Initial commit of Journaled-String-Tree

### PR DESCRIPTION
**Pull request by rrahn**
Adds several classes and functionality to create a Journaled-String-Tree.
Adds tests and first API documentation.

### CODE REVIEW

**Comment by rrahn** [comment]
@esiragusa @h-2: Can some one review this please? Maybe together, since it contains now a lot of changes.
Cheers

File: include/seqan/sequence_journaled/sequence_journaled.h [hunk 1]
```diff
@@ -526,6 +526,16 @@ setHost(String<TValue, Journaled<THostSpec, TJournalSpec, TBufferSpec> > & journ
     reinit(journaledString._journalEntries, length(str));
 }
 
+template <typename TValue, typename THostSpec, typename TJournalSpec, typename TBufferSpec, typename TSequence2>
```
Comment (esiragusa) [inline include/seqan/sequence_journaled/sequence_journaled.h]: Try to remove this one and the generic ```setHost(T1 & a, T2 const & b)```

File: tests/journaled_string_tree/testConfig.txt [hunk 2]
```diff
@@ -0,0 +1,28 @@
+!ConfigFile
```
Comment (esiragusa) [inline tests/journaled_string_tree/testConfig.txt]: Use VCF here

File: tests/journaled_string_tree/test_delta_map.cpp [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    SEQAN_CALL_TEST(test_delta_map_delta_store_deletion_size);
+    SEQAN_CALL_TEST(test_delta_map_delta_store_insertion_size);
+    SEQAN_CALL_TEST(test_delta_map_delta_store_net_size);
+
+    // Tests for delta map.
+    SEQAN_CALL_TEST(test_delta_map_insert);
+    SEQAN_CALL_TEST(test_delta_map_erase);
+    SEQAN_CALL_TEST(test_delta_map_find);
+    SEQAN_CALL_TEST(test_delta_map_size);
+    SEQAN_CALL_TEST(test_delta_map_empty);
```
Comment (esiragusa) [inline tests/journaled_string_tree/test_delta_map.cpp]: Test clear()

Comment (esiragusa) [inline tests/journaled_string_tree/test_delta_map.cpp]: Add swap()

File: tests/journaled_string_tree/test_journaled_string_tree.cpp [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+#include <seqan/basic.h>
+#include <seqan/file.h>
+
+#include "test_journaled_string_tree.h"
+
+SEQAN_BEGIN_TESTSUITE(test_journaled_string_tree)
+{
+    // Tests for journaled string tree
+    SEQAN_CALL_TEST(test_journaled_string_tree_constructor);
```
Comment (esiragusa) [inline tests/journaled_string_tree/test_journaled_string_tree.cpp]: Immutable JST is ImmutableStringSetConcept

Comment (esiragusa) [inline tests/journaled_string_tree/test_journaled_string_tree.cpp]: Check StrinSetConcept interface

Comment (esiragusa) [inline tests/journaled_string_tree/test_journaled_string_tree.cpp]: Should dimension() be length()?

Comment (esiragusa) [inline tests/journaled_string_tree/test_journaled_string_tree.cpp]: Should branch length be moved from JST to its graph iterator?

File: tests/journaled_string_tree/test_journaled_string_tree.h [hunk 5]
```diff
@@ -0,0 +1,327 @@
...
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ==========================================================================
+// Author: Rene Rahn <rene.rahn@fu-berlin.de>
+// ==========================================================================
+// Tests for the journaled string tree.
+// ==========================================================================
+
+#ifndef EXTRAS_TESTS_JOURNALED_STRING_TREE_TEST_JOURNALED_STRING_TREE_H_
```
Comment (esiragusa) [inline tests/journaled_string_tree/test_journaled_string_tree.h]: Remove prefix EXTRAS_

File: include/seqan/journaled_string_tree/observable.h [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+// ============================================================================
+// Forwards
+// ============================================================================
+
+// ============================================================================
+// Tags, Classes, Enums
+// ============================================================================
+
+template <typename TObserver, typename TList = String<TObserver*> >
+class Observable
```
Comment (esiragusa) [inline include/seqan/journaled_string_tree/observable.h]: class Observable<T> : public T {};

**Comment by rrahn** [comment]
@esiragusa I finished now integrating all the changes we discussed so far.

**Comment by esiragusa** [comment]
Okay then merge it :+1:

**Comment by esiragusa** [comment]
@rrahn Oh it seems to me that build_dox is broken... :-1:

**Comment by rrahn** [comment]
Thanks for pointing this out. I will fix it

**Pull request merged**